engine/language-server/src/test/scala/org/enso/languageserver/libraries/ComponentGroupsResolverSpec.scala

```diff
@@ -0,0 +1,258 @@
+        libraryComponentGroup("Foo", "Quux", "Mod2", "one")
+      )
+    }
+
+    "drop duplicated library definitions" in {
+      val resolver = new ComponentGroupsResolver
+      val testPackages = Vector(
+        config("Foo", "Bar"),
+        config(
+          "Foo",
+          "Baz",
+          ComponentGroups(
+            List(newComponentGroup("Mod1", "one", "two")),
+            List()
+          )
+        ),
+        config(
+          "Foo",
+          "Baz",
+          ComponentGroups(List(newComponentGroup("Mod1", "one")), List())
+        ),
+        config(
+          "Foo",
+          "Quux",
+          ComponentGroups(List(newComponentGroup("Mod2", "one")), List())
+        )
+      )
+
+      resolver.run(testPackages) shouldEqual Vector(
+        libraryComponentGroup("Foo", "Baz", "Mod1", "one", "two"),
+        libraryComponentGroup("Foo", "Quux", "Mod2", "one")
+      )
+    }
+
+    "drop duplicated component group definitions" in {
+      val resolver = new ComponentGroupsResolver
+      val testPackages = Vector(
+        config("Foo", "Bar"),
+        config(
+          "Foo",
+          "Baz",
+          ComponentGroups(
+            List(
+              newComponentGroup("Mod1", "one", "two"),
+              newComponentGroup("Mod1", "three")
+            ),
+            List()
+          )
+        ),
+        config(
+          "Foo",
+          "Quux",
+          ComponentGroups(List(newComponentGroup("Mod2", "one")), List())
+        )
+      )
+
+      resolver.run(testPackages) shouldEqual Vector(
+        libraryComponentGroup("Foo", "Baz", "Mod1", "one", "two"),
+        libraryComponentGroup("Foo", "Quux", "Mod2", "one")
+      )
+    }
+
+    "apply extended component groups" in {
+      val resolver = new ComponentGroupsResolver
+      val testPackages = Vector(
+        config(
+          "user",
+          "Unnamed",
+          ComponentGroups(
+            List(newComponentGroup("Main", "main")),
+            List(
+              extendedComponentGroup("Standard", "Base", "Data.Vector", "quux")
+            )
+          )
+        ),
+        config(
+          "Standard",
+          "Base",
+          ComponentGroups(
+            List(newComponentGroup("Data.Vector", "one", "two")),
+            List()
+          )
+        ),
+        config(
+          "user",
+          "Vector_Utils",
+          ComponentGroups(
+            List(),
+            List(
+              extendedComponentGroup("Standard", "Base", "Data.Vector", "three")
+            )
+          )
+        )
+      )
+
+      resolver.run(testPackages) shouldEqual Vector(
+        libraryComponentGroup("user", "Unnamed", "Main", "main"),
+        libraryComponentGroup(
+          "Standard",
+          "Base",
+          "Data.Vector",
+          "one",
+          "two",
+          "quux",
+          "three"
+        )
+      )
+    }
+
+    "skip component groups extending nothing" in {
```

Comment:
Again, shouldn't this be reported somewhere as a warning?

If I make a typo in specifying what existing group I want to extend, my extensions will just not show up - not very helpful when trying to figure out what was the issue. Not saying this is necessary from the beginning, but at least an indication that it should be added at some point would IMO be good.